salt/engines/slack.py

```diff
@@ -82,10 +105,11 @@ def _get_users(token):
 
 def start(token,
           aliases=None,
-          valid_users=None,
-          valid_commands=None,
+          valid_users=[],
```

Comment:
We don't allow mutable values as kwarg defaults. Here's a good explanation as to why: http://reinout.vanrees.org/weblog/2012/04/18/default-parameters.html

Instead, our typical approach is to set `None` as the default and then switch it over inside the function implementation to a list. Let me know if you have questions.